src/views/test/OperationsCustomers.tsx

```diff
@@ -0,0 +1,144 @@
+export const sortCustomers = (sort, onSortCustomers) => {
+    const sortModel: object[] = sortModelFromDataGrid(sort)
+    if (sortModel.length > 0) {
+        setSortQuery(sort, onSortCustomers)
+    } else {
+        unsortCustomers(onSortCustomers)
+    }
+}
+
+const setSortQuery = (sort, onSortCustomers) => {
+    onSortCustomers(sortQueryFromGridData(sort))
+}
+
+const sortQueryFromGridData = (sort): object => {
+    const sortQuery: object = {}
+    const sortModels = sortModelFromDataGrid(sort)
+    sortModels.forEach(sortModel => {
+        const sortColumnFiled: string = sortModel?.field
+        const sortValue: string = sortModel?.sort
+        sortQuery[sortColumnFiled] = sortValue
+    })
+    return sortQuery
+}
+
+const unsortCustomers = (onSortCustomers) => {
+    onSortCustomers({
+        "name": "asc"
+    })
+}
+
+const sortModelFromDataGrid = (sort) => {
+    const sortModel = sort?.sortModel
+    if (sortModel)
+        return sortModel
+    return {}
+}
+
+//TODO date
+// export const customerListDate = (customers) => {
+//     const customerListDate = customers.filter((date) => date.createdAt.includes("2021-03-17"))
+//     console.log(customerListDate,'customers'); 
+//     if (customerListDate?.length > 0) {
+//       return customerListDate[0]?.createdAt
+//     }
+// } 
+// console.log(customerListDate, 'customerListDate');
+
+export const pageByTotalAndPageSize = (pageSizeNumber: number, totalCustomers: number): number => {
+    const pageNumber: number = Math.floor((totalCustomers / pageSizeNumber) - 1)
+    console.log(pageNumber,'pageNumber'); 
+    if ((pageNumber) <= 0) {
+        return 0
+    }
+    return pageNumber
+}
+
+export const filterDataGrid = (filter, onFilterCustomers, onChangePageCustomers) => {
+    const filteredQueryForGraphQl = getQueryFromDataGrid(filter)
+    sendFilterQueryToGraphQl(filter, filteredQueryForGraphQl, onFilterCustomers)
+    setCurrentPageToOne(onChangePageCustomers)
+}
+
+const getQueryFromDataGrid = (filter) => {
+    const filteredQueryForGraphQl: object = {}
+    const filterModels = filterModelFromDataGrid(filter)
+
+    filterModels.forEach(filterModel => {
+        const filterColumnField: string = filterModel.columnField
+        const filterOperator: string = filterModel.operatorValue
+        const filterValue: string = filterModel.value
+        if (filterOperator === 'contains') {
```

Comment:
Let's support all operators that can come from Material-ui (you need to use different data types for columns).
Extract the logic that maps Material-ui operators to Hasura operators to a reusable function.